### ISSUE TITLE
How can i catch the exception " Failed to pin Bitmap "

### ISSUE THREAD

**Issue by UFreedom** [facebook/fresco#1862]
I know this exception due to an invalid image.But I can't make sure the picture is valid before using Fresco to load the picture,

When the picture goes wrong,I don't want to my app stoped because of the exception throw, I want my app to be continue.

Also, Fresco can configure failureImage and, as I understand, when the picture is invalid, Fresco can help me show failureImage instead of throwing a runtime exception.

If Fresco must throw the "Failed to pin Bitmap" exception, How can I catch it?

Hear is a stack trace for this exception:

```java
Fatal Exception: java.lang.RuntimeException: Failed to pin Bitmap
       at com.facebook.imagepipeline.nativecode.Bitmaps.nativePinBitmap(SourceFile)
       at com.facebook.imagepipeline.nativecode.Bitmaps.pinBitmap(SourceFile:44)
       at com.facebook.imagepipeline.platform.DalvikPurgeableDecoder.pinBitmap(SourceFile:155)
       at com.facebook.imagepipeline.platform.KitKatPurgeableDecoder.pinBitmap(SourceFile:32)
       at com.facebook.imagepipeline.platform.DalvikPurgeableDecoder.decodeJPEGFromEncodedImage(SourceFile:90)
       at com.facebook.imagepipeline.platform.KitKatPurgeableDecoder.decodeJPEGFromEncodedImage(SourceFile:32)
       at com.facebook.imagepipeline.bitmaps.HoneycombBitmapFactory.createBitmapInternal(SourceFile:65)
       at com.facebook.imagepipeline.bitmaps.PlatformBitmapFactory.createBitmap(SourceFile:79)
       at com.facebook.imagepipeline.bitmaps.PlatformBitmapFactory.createBitmap(SourceFile:46)
       at com.facebook.fresco.animation.bitmap.preparation.DefaultBitmapFramePreparer$FrameDecodeRunnable.prepareFrameAndCache(SourceFile:146)
       at com.facebook.fresco.animation.bitmap.preparation.DefaultBitmapFramePreparer$FrameDecodeRunnable.prepareFrameAndCache(SourceFile:164)
       at com.facebook.fresco.animation.bitmap.preparation.DefaultBitmapFramePreparer$FrameDecodeRunnable.run(SourceFile:114)
       at com.facebook.common.executors.ConstrainedExecutorService$Worker.run(SourceFile:178)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at com.facebook.imagepipeline.core.PriorityThreadFactory$1.run(SourceFile:43)
       at java.lang.Thread.run(Thread.java:841)
```

**Comment by erikandre** [comment]
Hi @UFreedom, could you please provide a stack trace for this exception?

**Comment by UFreedom** [comment]
@erikandre I have update my issue and provide the stack trace.

Thanks~

**Comment by erikandre** [comment]
@UFreedom, sorry for taking a while to respond.
Looks like there is no easy way to catch this exception without modifying Fresco itself. Probably we should never crash if this error occurs, it would be better to fail safely instead.

**Comment by wangrunxiang** [comment]
i have same one

**Comment by geetgobindsingh** [comment]
Is there any update with this issue ?

**Comment by erikandre** [comment]
@geetgobindsingh, @UFreedom a fix for this will be included in the upcoming release (planned for the next few days).

**Comment by haodadaa** [comment]
@erikandre is this bug fix in the last version(1.8.0)?

**Comment by erikandre** [comment]
Yes @haodadaa, the fix is in 1.8.0
